src/frontend/apps/e2e/__tests__/app-impress/language.spec.ts

```diff
@@ -25,4 +25,25 @@ test.describe('Language', () => {
       }),
     ).toBeVisible();
   });
+
+  test('checks the language cookie', async ({ page, context }) => {
+    // check initial user language detected and set as cookie
+    const initial_cookies = await context.cookies();
+    const initial_language_cookie = initial_cookies.find(
+      (cookie) => cookie.name == 'impress_language',
+    );
+    await expect(initial_language_cookie?.value).toBe('en');
+
+    // switch user language to fr
+    const header = page.locator('header').first();
+    await header.getByRole('combobox').getByText('English').click();
+    await header.getByRole('option', { name: 'Français' }).click();
+
+    // check if cookie is updated
+    const cookies = await context.cookies();
+    const language_cookie = cookies.find(
+      (cookie) => cookie.name == 'impress_language',
+    );
+    await expect(language_cookie?.value).toBe('fr');
+  });
```

Comment:
I think we should improve this test, it could be nice to assert that the backend is switching from 1 language to another.
If you go to this url (http://localhost:3000/docs/something/) you will generate a backend 404 error, you can check the error message to assert that the backend is synch with the frontend.

To intercept response with playwright you can check this test as an example:
https://github.com/rvveber/impress/blob/0cfcb84139452ffff9ba243f1e660bc33e1d5020/src/frontend/apps/e2e/__tests__/app-impress/doc-member-create.spec.ts#L11-L15
 
To redirect to a special page, you can do: `await page.goto('/docs/something/');`

Reply:
Thanks for the instructions!